common/models/role.js

```diff
@@ -244,31 +256,32 @@ module.exports = function(Role) {
         return;
       }
       debug('Model found: %j', inst);
-      var ownerId = inst.userId || inst.owner;
-      // Ensure ownerId exists and is not a function/relation
-      if (ownerId && 'function' !== typeof ownerId) {
-        if (callback) callback(null, matches(ownerId, userId));
-        return;
```

Comment:
I assume that you already looked at my [comment here](https://github.com/strongloop/loopback/pull/3140#issuecomment-277457162) and you prefer to keep this behavior in loopback 3 despite the comments.

So unless further comments i'll put back this behavior in place when `principalType === 'USER'`